pkg/sql/schemachanger/scrun/scrun.go

```diff
@@ -89,6 +89,14 @@ func runTransactionPhase(
 		return scpb.CurrentState{}, jobspb.InvalidJobID, nil
 	}
 	stages := sc.StagesForCurrentPhase()
+	if len(stages) == 0 {
```

Comment:
This smells. Can it really happen? Maybe we ought to detect this case and make an op that is equivalent to your undo op.

Reply:
Yes it can happen if there are dep rules which prevent the side-effects from happening for some reason. Dealing with in-stmt-txn side effects properly is a whole problem in itself which I feel is out of scope of this change and deserves proper treatment.

Reply:
I feel that this is an acceptable state of affairs for the time being, it's not less-worse than what we had previously nor is there an obvious solution at hand.